app/Core/HttprStatic.ts

```diff
@@ -34,12 +35,19 @@ export class HttprStatic {
       params: params || {}
     };
 
-    if (typeof body === 'object') {
-      settings.headers[HttpHeaders.CONTENT_TYPE] = MediaTypes.APPLICATION_JSON;
-      settings.body = JSON.stringify(body);
-    } else if (body) {
-      settings.headers[HttpHeaders.CONTENT_TYPE] = MediaTypes.TEXT_PLAIN;
-      settings.body = body;
+    if (body) {
+      if (body instanceof HttprBody) {
+        settings.headers[HttpHeaders.CONTENT_TYPE] = body.type;
+        settings.body = body.toString();
+      } else if (typeof body === 'object') {
```

Comment:
Maybe fallback to JSON content type should be after checking if headers contain content-type